File: ruby/selectionsort.rb
def selection_sort(lista)
  n = lista.length

  (n-1).times do |i|
    min = i

    (i + 1).upto(n) do |j|
      min = j if lista[j] < lista[min]
    end 

    lista[i], lista[min] = lista[min], lista[i] if min != i
  end

  lista
end

a = [9, 2, 6, 4, 1, 0, 3, 7, 5, 8]
p selection_sort(a)

File: go/selectionsort.go
package main

import (
	"fmt"
)

func main() {
	lista := []int{9, 2, 6, 4, 1, 0, 3, 7, 5, 8}
	fmt.Println("Antes da ordenação: ")
	fmt.Println(lista)

	selectionSort(lista)
	fmt.Println("Depois da ordenação: ")
	fmt.Println(lista)
}

func selectionSort(items []int){
	var n = len(items);
	for i := 0; i < n; i++ {
		var minIdx = i;
		for j := i + 1; j < n; j++ {
			if items[minIdx] > items[j] {
				minIdx = j;
			}
		}

		items[i], items[minIdx] = items[minIdx], items[i];
	}
}